### PR TITLE
Use multihost API with single host

### PR DESCRIPTION
**Pull request by petrasovaa**
Update to latest pops-core and use new API inside pops.cpp using multi-host with a single host.

### CODE REVIEW

**Comment by petrasovaa** [comment]
@ChrisJones687 This seems to be finally working, right now the pops_model_cpp has the same API, just uses the new pops-core API. Do you want to merge this first and continue in a separate PR with the actual implementation of multi-host that would modify the API, or continue here?

**Comment by ChrisJones687** [comment]
Yes, let's merge this and then open a new commit for the rest of this.

**Comment by petrasovaa** [comment]
Weird, looks like tests on mac os ended with some errors. Any idea?

**Comment by ChrisJones687** [comment]
That is weird. It has never failed those tests before. That makes it seem like something has changed. I will pull it and run it locally to see what is happening.

**Comment by ChrisJones687** [comment]
This fails locally for me as well. I don't think this means that the changes are incorrect just that for some reason the way the random seeds are being used here has changed the results for the specific random seed being used for these tests. In hindsight, these aren't the greatest tests because they can fail if a specific random draw has establishment occur early. This suggests that what we did changes the order of the stochastic processes. 

As a note, these tests pass for other random seeds I have tried.

**Comment by ChrisJones687** [comment]
That is a good point that there has to be a change to how MacOS does random number generation since it passes elsewhere else and has historically worked with MacOS. 

True on for the expectation of the tests as these are meant to test for explicit results for that seed and have always worked well before.

File: src/pops.cpp [hunk 1]
```diff
@@ -218,16 +221,87 @@ List pops_model_cpp(
...
-            treatment_maps[t],
-            pops::Date(treatment_dates[t]),
-            pesticide_duration[t],
-            treatment_application);
-        config.use_treatments = true;
+      treatments.add_treatment(
+        treatment_maps[t],
+                      pops::Date(treatment_dates[t]),
+                      pesticide_duration[t],
+                                        treatment_application);
```
Comment (wenzeslaus) [inline src/pops.cpp:233]: Inconsistent indent.

**Review by wenzeslaus — APPROVED** [review]
Looking just at the code and ignoring the failing tests, the code itself looks good except the little issue with indentation.

**Comment by petrasovaa** [comment]
Looks like the tests on Mac are fixed!

**Comment by petrasovaa** [comment]
@ChrisJones687 one more complication, the tests fail when the host uncertainty is used. Vashek added this check https://github.com/ncsu-landscape-dynamics/pops-core/blob/main/include/pops/multi_host_pool.hpp#L365
to avoid probabilities > 1, and since the randomly generated hosts go above total population, the tests fail.

**Comment by ChrisJones687** [comment]
I pushed a check that ensures that hosts are never > than total population but it still fails. It is always like 1.002 or 1.005 which suggests to me that it is a rounding issue in pops-core. Thoughts?

**Pull request merged**